core/rpc/client/user/jsonrpc/methods.go

```diff
@@ -212,10 +212,16 @@ func (cl *Client) TxQuery(ctx context.Context, txHash []byte) (*transactions.TcT
 	if err != nil {
 		return nil, err
 	}
+
+	var tx transactions.Transaction
+	if res.Tx != nil {
+		tx = *res.Tx
+	}
+
 	return &transactions.TcTxQueryResponse{
 		Hash:     res.Hash,
 		Height:   res.Height,
-		Tx:       *res.Tx,
+		Tx:       tx,
```

Comment:
Looks like we should make this field a pointer.
Seems sloppy to return the zero value and all the nil and undefined fields of this type.

Reply:
I agree with that